src/ItemRenderer.js

```diff
@@ -59,6 +59,8 @@ export default class ItemRenderer extends PureComponent<Props, void> {
           key={uid}
           label={node.name}
           onClick={() => itemData.focusNode(uid)}
+          onMouseOut={onMouseOut ? e => onMouseOut(e, node.source) : null}
+          onMouseOver={onMouseOver ? e => onMouseOver(e, node.source) : null}
```

Comment:
I don't think this messes with React's shouldComponentUpdate as we're already implementing that via the PureComponent here, but this may be re-rendering more than necessary.